### PR TITLE
[PROPIN-558] Brand New Highlight Section

### PR DESCRIPTION
**Pull request by gabrielagabriel**
## Description

https://www.loom.com/share/d65eef685bb24aa3a8c238f14ed39238

## What has been done

- Export styled-components to be used inside Section

## Things to consider

**This is a specialized component. It has a closed API. il only serves to highlight trip/rides**

- This component uses Grid 'cause is easier to declare than creating grids with flexbox

## How it was tested

- RTL

### **A11Y**
@nickoola  With A11y review
https://www.loom.com/share/418ec414ddef4f599652230fbb146624?sharedAppSource=personal_library

### CODE REVIEW

**Comment by mcellier** [comment]
nit: If it's only for trips, put that information in the name. HighlightSection might be too generic. HighlightedRidesSection?

File: src/layout/section/highlightSection/HighlightSection.tsx [hunk 1]
```diff
@@ -1,27 +1,67 @@
...
+import { ItemChoiceProps } from '../../../itemChoice'
+import { Col, Grid, HighlightSectionElements } from './HighlightSection.style'
 
+type ContentItemsType = ItemChoiceProps &
+  Readonly<{
+    hidden: boolean
+  }>
+
+type GridListItemsProps = { items: Array<ContentItemsType> }
+export const GridListItems = ({ items }: GridListItemsProps) => {
```
Comment (mcellier) [inline src/layout/section/highlightSection/HighlightSection.tsx:12]: not need to export them IMHO

**Review by Tuxosaurus — APPROVED** [review]

**Comment by gabrielagabriel** [comment]
> @mcellier nit: If it's only for trips, put that information in the name. HighlightSection might be too generic. HighlightedRidesSection?

I agree, but I don't wanna rename the component.

**Comment by nickoola** [comment]
@gabrielagabriel did you navigate with the keyboard only & by using a screen reader?
Especially for the mechanism to hide/show content.

**Comment by nickoola** [comment]
An other remark, list items should be links `<a>` instead of `<button>`:
Screenshot 2021-02-17 at 08 22 02

**Review by nickoola — APPROVED** [review]

**Comment by nickoola** [comment]
Semantic & A11y part is good for me 👌 
Great job 👍

File: src/layout/section/highlightSection/HighlightSection.tsx [hunk 2]
```diff
@@ -1,27 +1,100 @@
...
+  const { axes, destinations } = highlights
+
+  const displayedItems = axes.items.map((item, index) => ({
+    ...item,
+    hidden: collapsed && index >= DEFAULT_ITEMS_SIZE,
+  }))
+
+  // Set Focus
+  const [collapsibleRegionWrapper] = setFocus(collapsed)
+  const collapsibleRegionId = uniqueId('region-')
```
Comment (Tuxosaurus) [inline src/layout/section/highlightSection/HighlightSection.tsx:70]: Thanks, I forgot the `uniqueId` existed, will need to use it more

File: src/layout/section/highlightSection/HighlightSection.unit.tsx [hunk 3]
```diff
@@ -1,22 +1,190 @@
 import React from 'react'
-import renderer from 'react-test-renderer'
 
+import { fireEvent, render } from '@testing-library/react'
+
+import { RideAxis } from '../../../_utils/rideAxis'
 import { HighlightSection } from './index'
 
+export const axes = [
```
Comment (Tuxosaurus) [inline src/layout/section/highlightSection/HighlightSection.unit.tsx:8]: Are these exports needed (same in story)?

File: src/layout/section/highlightSection/HighlightSection.tsx (hunk 2 above)
Comment (gabrielagabriel) [reply src/layout/section/highlightSection/HighlightSection.tsx:68]: yeah! it's a great thing to compose our lists!

**Pull request merged**